forwards_paper.tex

```diff
@@ -854,35 +871,45 @@ \subsubsection*{Computational complexity}
 Similarly, not every new edge derives from a never-before-seen node,
 but the number of nodes is at most equal to the number of edges plus the sample size.
 The space complexity after simplification is therefore $O(N + N\log N)$, since
-we need $O(N)$ space to store the first marginal tree, and $O(1)$ space
+we need $O(N)$ space to store the first marginal tree, and a constant amount of space
 for each of the $O(N \log N)$ trees that follow. Thus, we need $O(N^2)$ space
-to store the complete history of the simulation, but only $O(N + N \log N)$
+to store the complete history of the simulation, but only $O(N \log N)$ % $O(N + N \log N)$
 space to store the history that is relevant to the extant population.
 
 %% Progressive memory reduction as a function of T
 \citet{padhukasahasram2008exploring} obtained impressive speedups relative to contemporary simulators
 by retaining only 8 generations of history.
 This begs the question: how fast does the information in the pedigree become irrelevant?
-Suppose instead that we have run a simulation of $2N$ individuals for $T$ generations,
+Suppose instead that we have run a simulation of $N$ individuals for $T$ generations,
 begun with no prior history.
 One way to answer this question is to ask:
-How many of the resulting $4NT$ edges are required after simplification?
-As above, the expected number of edges is bounded by $2N-2$ plus four times the mean marginal tree length.
-Roughly, the length of time for which a marginal tree has $k$ tips is
-$2N/(k(k-1)) = 2N(1/(k-1) - 1/k)$ generations,
-and so the time to go from $N$ to $n$ tips is $2N(1/(n-1) - 1/(N-1))$.
-This implies that after running for time $T$ we expect to
-have around $n(T)$ roots, where $n(T) \approx 1 + 2N/(T+2)$.
+How many of the resulting $2NT$ edges are required after simplification?
+As above, the expected number of edges is bounded by $N-2$
+plus four times the sum of the lengths of all marginal trees,
+because new edges appear with most recombinations that occur on any marginal tree.
+Therefore, we need to compute the mean total length of all branches in a coalescent tree
+up until time $T$ (or the common ancestor, whichever comes first).
+Since $k$ lineages coalesce at rate $\binom{k}{2}/N$,
+the expected length of time for which a marginal tree has $k$ branches is
```

Comment:
Shouldn't this be k roots rather than branches? We have k extant lineages/roots in the tree and they coalesce at rate k choose 2, but the number of branches in the marginal tree is more complicated, isn't it?

Reply:
I mean that if you are following the coalescent process up, then $k$ lineages coalesce at rate $k$ choose 2.  But then I waffled on whether to call these things "lineages" or "roots" or "branches".  I think that "lineages" is best, perhaps.  But then, we haven't explained what a lineage is, or indeed what "coalesce at rate" means.  I guess my goal here was to make this understandable to someone with a basic understanding of coalescent theory, in which case perhaps we don't need to define lineage.  What do you think about that strategy?  Options:

1. assume basic understanding of coal theory, along current lines
2. not assume this, and insert more explanation
3. not assume this and remove most of the arguments, just referring to "coalescent theory calculations tells us that X"

Reply:
Actually, reading this again, I think it's right as stated: we're talking about the period of time over which there are `k` branches in the tree (thinking about the tree, not the coalescent process).  To figure this out, we ask about the time that the coalescent process has `k` lineages.  The questions there stand, though.

Reply:
OK cool. I'd vote for option (1)